sdmetrics/reports/single_table/_properties/column_pair_trends.py

```diff
@@ -0,0 +1,441 @@
+            if column_sdtype in ['numerical', 'datetime']:
+                name_discrete = create_unique_name(column_name + '_discrete', metadata['columns'])
+                processed_data[name_discrete], bin_edges = self._discretize_column(
+                    column_name, processed_data[column_name]
+                )
+
+        return processed_data
+
+    def _get_metric(self, sdtype_col_1, sdtype_col_2):
+        """Get the metric to use for the property.
+
+        If one of the columns is discrete, use the ContingencySimilarity metric.
+        Otherwise, use the CorrelationSimilarity metric.
+
+        Args:
+            sdtype_col_1 (str):
+                The sdtype of the first column
+            sdtype_col_2 (str):
+                The sdtype of the second column
+        """
+        if self._sdtype_to_shape[sdtype_col_1] != self._sdtype_to_shape[sdtype_col_2]:
+            metric = ContingencySimilarity
+        elif self._sdtype_to_shape[sdtype_col_1] == 'continuous':
+            metric = CorrelationSimilarity
+        else:
+            metric = ContingencySimilarity
+
+        return metric
+
+    def _get_columns_data(self, column_name_1, column_name_2, data, metadata):
+        """Get the data for the property.
+
+        If one is comparing a continuous column to a discrete column, use the discrete version
+        of the continuous column. Otherwise use the original columns.
+
+        Args:
+            column_name_1 (str):
+                The name of the first column
+            column_name_2 (str):
+                The name of the second column
+            data (pandas.DataFrame):
+                The data
+            metadata (dict):
+                The metadata of the table
+        """
+        sdtype_col_1 = metadata['columns'][column_name_1]['sdtype']
+        sdtype_col_2 = metadata['columns'][column_name_2]['sdtype']
+        col_name_1 = column_name_1
+        col_name_2 = column_name_2
+        if self._sdtype_to_shape[sdtype_col_1] != self._sdtype_to_shape[sdtype_col_2]:
+
+            if self._sdtype_to_shape[sdtype_col_1] == 'continuous':
+                col_name_1 = create_unique_name(column_name_1 + '_discrete', metadata['columns'])
+            else:
+                col_name_2 = create_unique_name(column_name_2 + '_discrete', metadata['columns'])
+
+        return data[[col_name_1, col_name_2]]
+
+    def _required_preprocessing(self, sdtype_col_1, sdtype_col_2):
+        """Check if a processing of one of the columns was required to compute the metric.
+
+        Args:
+            sdtype_col_1 (str):
+                The sdtype of the first column
+            sdtype_col_2 (str):
+                The sdtype of the second column
+        """
+        if sdtype_col_1 == 'datetime' or sdtype_col_2 == 'datetime':
+            return True
+        elif self._sdtype_to_shape[sdtype_col_1] != self._sdtype_to_shape[sdtype_col_2]:
+            return True
+        else:
+            return False
```

Comment:
Instead of doing all this, I propose just storing two lists:
1. Of column names that failed during datetime
2. Of column names that failed during discretizing
Then when we try to run the metric, we can just see if either column is in the list

Reply:
Updated in [ebe9e54](https://github.com/sdv-dev/SDMetrics/pull/368/commits/ebe9e54c38ed06f78a6c4ca1a0bf16eb524501e7)